src/plugins/embeddable/public/lib/panel/panel_header/panel_actions/customize_panel/customize_panel_editor.tsx

```diff
@@ -125,8 +125,14 @@ export const CustomizePanelEditor = (props: CustomizePanelProps) => {
             <EuiButtonEmpty
               size="xs"
               data-test-subj="resetCustomEmbeddablePanelTitleButton"
-              onClick={() => setPanelTitle(embeddable.getOutput().defaultTitle)}
-              disabled={hideTitle || !editMode}
+              onClick={() => {
+                setPanelTitle(embeddable.getInput().title ?? embeddable.getOutput().defaultTitle);
+              }}
+              disabled={
+                hideTitle || !editMode || embeddable.getInput().title != undefined
+                  ? embeddable.getInput().title === panelTitle
+                  : embeddable.getOutput().defaultTitle === panelTitle || panelTitle === ''
+              }
```

Comment:
This should only be disabled if the panel title is equal to the default title:

```suggestion
              disabled={
                hideTitle || !editMode || embeddable.getOutput().defaultTitle === panelTitle
              }
```